### DISCUSSION TITLE
Ways to highlight most interesting phrases in a text using sentiment analysis

### DISCUSSION THREAD

**Discussion by george-i** [winkjs/wink-nlp#65] [Help on coding] [answered]
I'm looking at ways to highlight most interesting phrases in a text using sentiment analysis and the first way I thought I could do it is by identifying phrases which have the most scored words.  

Let's take this text as an example:  

> Modern card issuer Marqeta has announced its partnership with FNBO to expand its partner ecosystem and allow customers to launch modern credit card programmes. The new collaboration aims to modernise the credit card offering and meet clients’ demands, offering a more flexible and reliable digital experience. FNBO and Marqeta will allow companies to easily launch credit cards using the latter’s APIs and embed the card experience within their app ecosystem. **The Marqeta platform benefits from a self-service dashboard to update credit products according to clients ‘needs, while client companies can instantly extend credit applications, decisions, and onboard accounts, among other features**. First National Bank of Omaha is a subsidiary of First National of Nebraska, with offices in Nebraska, Colorado, Illinois, Kansas, and Texas, among others. It is a six-generation privately own BaaS. Marqeta is headquartered in California and is certified to operate and offer its flexible payment solutions in 36 countries globally.  

The bolded phrase has the most scored words, therefore it's highlighted as an interesting text to read.  
My question is if there other better ways to do this, because sometimes the highlighted text may seem out of context.   

And a second question, the text above has  
`{score: 8, normalizedScore: 0.3121}`  
Which of the two properties indicates better the text sentiment?

**Comment by sanjayaksaxena** [comment, marked as answer]
Please look at [wink-nlp](https://github.com/winkjs/wink-nlp), which is a super set of our existing nlp packages. An live demo of a similar scenario can be found at [How to perform sentiment analysis?](https://observablehq.com/@winkjs/how-to-perform-sentiment-analysis); this demo computes the sentiment score of every line of a poem and highlights with the color intensity proportional to the computed score. The [wink-nlp](https://github.com/winkjs/wink-nlp) has built in method to highlight either a sentence, entity or a token in any desired format (e.g. markdown or html). Here is the output if you replace the poem by the above text:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/29990/144849567-2f2b4f71-a0ca-4f5a-8ea1-e7ffe5f32095.png">
